doajtest/unit/test_models.py

```diff
@@ -1099,7 +1099,7 @@ def test_33_article_stats(self):
             articles.append(article)
 
         for i in range(len(articles)):
-            articles[i].save(blocking=i == len(articles) - 1)
+            articles[i].save(blocking=True)
```

Comment:
This could make the test quite a bit slower, and to solve this problem we did recently introduce an alternative approach.  Instead you can now do:

`Article.blockall([(a.id, a.last_updated) for a in articles])`

That will make sure that all articles are available in search without having to block each one individually.

Reply:
The problem here is that because they are new articles, there is no `last_updated` set. This causes the following exception:

```python
Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/steve/code/cl/doaj/doajtest/unit/test_models.py", line 1101, in test_33_article_stats
    models.Article.blockall([(a.id, a.last_updated) for a in articles])
  File "/home/steve/code/cl/doaj/portality/dao.py", line 731, in blockall
    cls.block(id, lu, sleep=sleep, max_retry_seconds=individual_max_retry_seconds)
  File "/home/steve/code/cl/doaj/portality/dao.py", line 707, in block
    threshold = datetime.strptime(last_updated, "%Y-%m-%dT%H:%M:%SZ")
TypeError: strptime() argument 1 must be string, not None
```

I've solved this by applying a last_updated of now():
```models.Article.blockall([(a.id, datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')) for a in articles])```

Reply:
Unless the missing save is applied. Oops. Incoming.